Ruff mistakes with Protocol and positional-only arguments.

I'm not sure if I've dug down to the root of the issue, but the following code incorrectly produces the following errors. When I remove the `/` and make `array` positional-or-keyword, rather than positional-only, the errors disappear.

```python
Array = TypeVar("Array")

class SomeCallable(Protocol):
    def __call__(self, array: Array, /, key: Any) -> Array:
        ...
```

```
[tool.ruff]
target-version = "py310"
select = ["ALL"]
```

```
>>> pre-commit run ruff -a

.../data.py:258:18: ANN001 Missing type annotation for function argument `self`
.../data.py:258:41: N805 First argument of a method should be named `self`
```

- [x] A minimal code snippet that reproduces the bug.
- [x] The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
- [x] The current Ruff settings (any relevant sections from your `pyproject.toml`).
- [x] The current Ruff version is "v0.0.243"

--- a/crates/ruff/resources/test/fixtures/flake8_annotations/annotation_presence.py
+++ b/crates/ruff/resources/test/fixtures/flake8_annotations/annotation_presence.py
@@ -103,3 +103,7 @@ class Foo:
     @classmethod
     def foo(cls, a: int, b: int) -> int:
         pass
+
+    # ANN101
+    def foo(self, /, a: int, b: int) -> int:
+        pass

--- a/crates/ruff/src/rules/flake8_annotations/rules.rs
+++ b/crates/ruff/src/rules/flake8_annotations/rules.rs
@@ -475,9 +475,9 @@ pub fn definition(
 
         // ANN001, ANN401
         for arg in args
-            .args
+            .posonlyargs
             .iter()
-            .chain(args.posonlyargs.iter())
+            .chain(args.args.iter())
             .chain(args.kwonlyargs.iter())
             .skip(
                 // If this is a non-static method, skip `cls` or `self`.
@@ -581,7 +581,7 @@ pub fn definition(
 
         // ANN101, ANN102
         if is_method && !visibility::is_staticmethod(checker, cast::decorator_list(stmt)) {
-            if let Some(arg) = args.args.first() {
+            if let Some(arg) = args.posonlyargs.first().or_else(|| args.args.first()) {
                 if arg.node.annotation.is_none() {
                     if visibility::is_classmethod(checker, cast::decorator_list(stmt)) {
                         if checker.settings.rules.enabled(&Rule::MissingTypeCls) {

--- a/crates/ruff_python_formatter/src/trivia.rs
+++ b/crates/ruff_python_formatter/src/trivia.rs
@@ -225,7 +225,6 @@ fn sorted_child_nodes_inner<'a>(node: &Node<'a>, result: &mut Vec<Node<'a>>) {
                 for decorator in decorator_list {
                     result.push(Node::Expr(decorator));
                 }
-                // TODO(charlie): Retain order.
                 for arg in &args.posonlyargs {
                     if let Some(expr) = &arg.node.annotation {
                         result.push(Node::Expr(expr));

